internal/fields/model.go

```diff
@@ -16,3 +16,52 @@ type FieldDefinition struct {
 	External    string            `yaml:"external"`
 	Fields      []FieldDefinition `yaml:"fields"`
 }
+
+func (orig *FieldDefinition) Update(fd FieldDefinition) {
+	if fd.Name != "" {
```

Comment:
Ok, so just have to remember about updating these conditions when adding a new property to the `FieldDefinition`.

I'm curious if it isn't shorter to use the YAML marshaller to update fields.

Reply:
> Ok, so just have to remember about updating these conditions when adding a new property to the `FieldDefinition`.
> 

Yes, but well, having it close to the struct I guess that this is easier.

> I'm curious if it isn't shorter to use the YAML marshaller to update fields.

I can give a try to this if prefered. Not sure if we will have flexibility on the nested fields, but we could keep this part as is.

Reply:
I don't want to slow you down with this as this is a working implementation. We can come back to this in the future.